src/components/Graph/MaxPrice.tsx

```diff
@@ -19,24 +21,33 @@ export interface LineChartProps {
 }
 
 export const MaxPrice = ({ label, yText, xScale, stroke, width, xDate }: LineChartProps) => {
-  const handleTextPos = (x: number): { x: number; anchor: 'end' | 'start' | 'middle' } => {
-    const offsetWidth = width / 2
-    const buffer = 16
-    const end = width - offsetWidth
-    if (x < offsetWidth) {
-      return { x: x + buffer, anchor: 'start' }
-    } else if (x > end) {
-      return { x, anchor: 'end' }
-    } else {
-      return { x, anchor: 'start' }
-    }
-  }
+  const makeTextPos = useCallback(
```

Comment:
Note: At this point this could be a pure function but I don't want to clutter this PR with more changes than necessary.